domain/application/service/unit.go

```diff
@@ -372,3 +373,83 @@ func (s *Service) GetUnitNamesForApplication(ctx context.Context, appName string
 	}
 	return names, nil
 }
+
+// GetPublicAddress returns the public address for the specified unit.
+// For k8s provider, it will return the first public address of the cloud
+// service if any, the first public address of the cloud container otherwise.
+// For machines provider, it will return the first public address of the
+// machine.
+//
+// The following errors may be returned:
+// - [uniterrors.UnitNotFound] if the unit does not exist
+func (s *Service) GetPublicAddress(ctx context.Context, unitName coreunit.Name) (network.SpaceAddress, error) {
+	appName, err := s.st.GetApplicationNameByUnitName(ctx, unitName)
+	if err != nil {
+		return network.SpaceAddress{}, errors.Capture(err)
+	}
+
+	serviceAddresses, err := s.st.GetCloudServiceAddresses(ctx, appName)
+	if err != nil {
+		return network.SpaceAddress{}, errors.Capture(err)
+	}
+	if len(serviceAddresses) > 0 {
+		if addr, ok := serviceAddresses.OneMatchingScope(network.ScopeMatchPublic); ok {
+			return addr, nil
+		}
+	}
+	// Fallback to the cloud container address.
+	containerAddresses, err := s.st.GetCloudContainerAddresses(ctx, unitName)
+	if err != nil {
+		return network.SpaceAddress{}, errors.Capture(err)
+	}
+	if len(containerAddresses) > 0 {
+		if addr, ok := containerAddresses.OneMatchingScope(network.ScopeMatchPublic); ok {
+			return addr, nil
+		}
+	}
+	// TODO(nvinuesa): We should also use the address taken from the
+	// machine table here when we migrate machines to dqlite.
+	return network.SpaceAddress{}, network.NoAddressError(string(network.ScopePublic))
+}
+
+// GetPrivateAddress returns the private address for the specified unit.
+// For k8s provider, it will return the first private address of the cloud
+// service if any, the first private address of the cloud container otherwise.
+// For machines provider, it will return the first private address of the
+// machine.
+//
+// The following errors may be returned:
+// - [uniterrors.UnitNotFound] if the unit does not exist
+func (s *Service) GetPrivateAddress(ctx context.Context, unitName coreunit.Name) (network.SpaceAddress, error) {
```

Comment:
Is there a reason for this to operate on app and unit names rather than UUIDs?

Reply:
On the service layer we have many of the methods that take names instead of UUIDs. We used this as a pattern in the application domain because they can be easily wired in the apiserver and retrieve the UUIDs in the service layer.